msg/service_test.go

```diff
@@ -39,3 +39,25 @@ func TestSplit255(t *testing.T) {
 		t.Fail()
 	}
 }
+
+func testDomainPath(t *testing.T, domain string, expected string) {
+	path := Path(domain)
+	if path != expected {
+		t.Logf("Path %s\n", domain)
+		t.Logf("%s %v\n", path, expected)
+		t.Fail()
+	}
+
+	roundtrip := Domain(path)
+	if roundtrip != domain + "." {
+		t.Logf("Domain %s\n", path)
+		t.Logf("%s %v\n", roundtrip, domain)
+		t.Fail()
+	}
+}
+
+
+func TestPath(t *testing.T) {
```

Comment:
you can just fold the testDomainPath into TestPath and rename TestPath to TestPathUnderscore. The tests can be stored in a string slice, where the even index as the domain string and the uneven index has the expected string.